### PR TITLE
Standardize usage of collections.abc.

### PR DESCRIPTION
**Pull request by bdice**
The codebase currently uses multiple ways of importing `collections.abc` classes in cudf. This can be problematic because the names in `collections.abc` can overlap with names in `typing`, so we need a way to disambiguate the two. This PR standardizes our imports such that abstract base classes follow a pattern so that `abc.` is always in the name of abstract base classes.

```python
from collections import abc
# Not "import collections.abc" or "from collections.abc import Mapping"

if isinstance(obj, abc.Mapping):
    pass
```

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -2997,10 +2998,10 @@ def agg(self, aggs, axis=None):
...
                     result[key] = getattr(col, value)()
-            elif all([isinstance(val, Iterable) for val in aggs.values()]):
+            elif all([isinstance(val, abc.Iterable) for val in aggs.values()]):
                 idxs = set()
                 for val in aggs.values():
-                    if isinstance(val, Iterable):
+                    if isinstance(val, abc.Iterable):
                         idxs.update(val)
                     elif isinstance(val, str):
                         idxs.add(val)
```
Comment (bdice) [inline python/cudf/cudf/core/dataframe.py:3007]: The order of these statements may be a bug. Strings meet the requirements of `Iterable`, so the string code path may never be taken here. Strings should probably be handled first. I'll fix this.

File: python/cudf/cudf/core/reshape.py [hunk 2]
```diff
@@ -485,14 +486,13 @@ def melt(
     1  b         B          3
     2  c         B          5
     """
-    assert col_level in (None,)
+    assert col_level is None
```
Comment (galipremsagar) [inline python/cudf/cudf/core/reshape.py:489]:
```suggestion
    if col_level is not None:
        raise NotImplementedError("col_level != None is not supported yet.")
```

We could change this assertion to a clear error message.

**Review by galipremsagar — COMMENTED** [review]
LGTM, minor comment.

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**